File: client/src/store/info.js
import axios from "axios";

const state = {
  loading: true,
  success: true,
  allowRegistration: true,
  loginEnabled: true,
  version: "loading",
  totalUsers: 1,
  name: "COCO Annotator",
  socket: null
};

const getters = {};

const mutations = {
  socket(state, connected) {
    state.socket = connected;
  },
  increamentUserCount(state) {
    state.totalUsers++;
  },
  getServerInfo(state) {
    state.loading = true;
    axios
      .get("/api/info/")
      .then(response => {
        state.loading = false;
        state.success = true;

        let data = response.data;
        state.version = data.git.tag;
        state.allowRegistration = data.allow_registration;
        state.loginEnabled = data.login_enabled;
        state.totalUsers = data.total_users;
      })
      .catch(() => {
        state.loading = false;
        state.success = true;

        state.version = "unknown";
      });
  }
};

const actions = {};

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
};

File: backend/webserver/api/info.py
from flask_restplus import Namespace, Resource

from workers.tasks import long_task
from config import Config
from database import UserModel, TaskModel


api = Namespace('info', description='Software related operations')


@api.route('/')
class Info(Resource):
    def get(self):
        """ Returns information about current version """

        return {
            "name": "COCO Annotator",
            "author": "Justin Brooks",
            "demo": "https://annotator.justinbrooks.ca/",
            "repo": "https://github.com/jsbroks/coco-annotator",
            "git": {
                "tag": Config.VERSION
            },
            "login_enabled": not Config.LOGIN_DISABLED,
            "total_users": UserModel.objects.count(),
            "allow_registration": Config.ALLOW_REGISTRATION
        }


@api.route('/long_task')
class TaskTest(Resource):
    def get(self):
        """ Returns information about current version """
        task_model = TaskModel(group="test", name="Testing Celery")
        task_model.save()

        task = long_task.delay(20, task_model.id)
        return {'id': task.id, 'state': task.state}